src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,117 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 /**
  * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
  * реалізації ArrayList (default capacity, newCapacity...)</p>
  */
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+    private int index;
+
+    public ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+    }
+
+    private void growArray() {
+        Object[] newArray = new Object[elements.length + elements.length / 2];
+        System.arraycopy(elements, 0, newArray, 0, elements.length);
+        elements = newArray;
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= this.index) {
+            throw new ArrayIndexOutOfBoundsException();
+        }
+    }
 
     @Override
     public void add(T value) {
+        if (index == elements.length) {
+            growArray();
+        }
```

Comment:
we can move this check inside this method

Reply:
moved